frontend/src/components/@common/Tag/Tag.tsx

```diff
@@ -0,0 +1,31 @@
+import { CloseRounded } from 'assets/icons';
+import { ComponentPropsWithoutRef, PropsWithChildren } from 'react';
+import { styled } from 'styled-components';
+
+type Props = {
+  removable?: boolean;
+} & ComponentPropsWithoutRef<'button'>;
+
+const Tag = ({ removable = true, children, ...rest }: Props) => {
+  return (
+    <S.Tag {...rest}>
+      #{children}
+      {removable && <CloseRounded width={14} height={14} />}
+    </S.Tag>
+  );
+};
+
+export default Tag;
```

Comment:
```suggestion
const Tag = ({ removable = true, children, ...rest }: Props) => {
  const [isShow, setIsShow] = useState(true);

  return (
    isShow && (
      <S.Tag {...rest} onClick={() => removable && setIsShow(false)}>
        #{children}
        {removable && <CloseRounded width={14} height={14} />}
      </S.Tag>
    )
  );
};
```
다음과 같이 `removable`이 `true` 일 때 내부적으로 태그를 삭제하도록 하면 어떨까요? 외부에서 삭제하는 로직을 받는게 살짝 어색하게 느껴져서 한번 제안해봅니다!

Reply:
음 생각은 해봤는데요
일단 제가 생각했을 때는 TagInput이라는 컴포넌트가 Tag 추가 삭제 수정 로직을 가지고 있고, Tag 자체는 그냥 글 정보에서도 사용할 수 있고 TagInput에서도 사용할 수 있는 공용 컴포넌트라는 생각을 했어요.

TagInput에서 모든 추가,삭제,수정 로직을 가지고 있기에 오히려 Tag에 삭제 로직만 있으니까 약간 로직이 분산되어 버리더라구요
아니면 오히려 여기서 removable prop과 Close 아이콘을 없애고 TagInput쪽에서 Close 아이콘을 같이 넣어주는 건 어떤가요?

```suggestion
const Tag = ({ children, ...rest }: Props) => {
  return (
    <S.Tag {...rest}>#{children}</S.Tag>
  );
};

export default Tag;
```

![image](https://github.com/woowacourse-teams/2023-dong-gle/assets/57815133/19344168-dd1a-4378-a7f3-52268726c956)

Reply:
음 그 이유라면 `removable` 일 때 삭제 아이콘만 추가시켜주는 기존 것도 좋아보입니다!